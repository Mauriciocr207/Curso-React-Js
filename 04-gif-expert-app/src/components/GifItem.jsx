import { string } from "prop-types";

export default function GifItem({ title, url }) {

  return (
    <div className="overflow-hidden relative" data-testid="GifItem">
      <div className="hover:opacity-75 opacity-0 transition-opacity duration-500 absolute overflow-y-auto p-2 bg-black w-full h-full rounded-lg flex">
        <p
          className="font-normal opacity-80 text-white text-center w-full p-2"
        >
          {title}
        </p>
      </div>
      <img
        className="h-full max-w-full rounded-lg object-cover object-center"
        src={url}
        alt={title}
      />
    </div>
  );
}

GifItem.propTypes = {
  title: string.isRequired,
  url: string.isRequired,
};
