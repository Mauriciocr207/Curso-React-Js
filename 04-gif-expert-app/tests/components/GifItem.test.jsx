import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";
import "@testing-library/jest-dom";

describe('Tests on <GifItem/>', () => {
  const title = "Gif de prueba";
  const url = "https://giphy.com/gifs/demon-slayer-zenitsu-thundergod-VEzYdo930nTiTuVeMU";

  test('match snapshot', () => {
    const { container } = render(<GifItem title={title} url={url}/>);
    expect(container).toMatchSnapshot();
  });

  test('must show image with correct alt and url', () => {
    render(<GifItem title={title} url={url}/>);
    const gifItem = screen.getByTestId('GifItem');
    const img = gifItem.querySelector('img');
    expect(img).toHaveAttribute("alt", title);
    expect(img).toHaveAttribute("src", url);
  })
})