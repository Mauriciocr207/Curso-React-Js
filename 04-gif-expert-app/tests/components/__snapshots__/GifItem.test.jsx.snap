// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem/> match snapshot 1`] = `
<div>
  <div
    class="overflow-hidden relative"
    data-testid="GifItem"
  >
    <div
      class="hover:opacity-75 opacity-0 transition-opacity duration-500 absolute overflow-y-auto p-2 bg-black w-full h-full rounded-lg flex"
    >
      <p
        class="font-normal opacity-80 text-white text-center w-full p-2"
      >
        Gif de prueba
      </p>
    </div>
    <img
      alt="Gif de prueba"
      class="h-full max-w-full rounded-lg object-cover object-center"
      src="https://giphy.com/gifs/demon-slayer-zenitsu-thundergod-VEzYdo930nTiTuVeMU"
    />
  </div>
</div>
`;
